src/x3f_image.c

```diff
@@ -115,9 +144,17 @@
 {
   uint32_t keep[4], keep_cols, keep_rows;
 
-  if (!x3f_get_camf_matrix(x3f, name, 4, 0, 0, M_UINT, rect)) return 0;
+  if (!strncmp(name, "Fake", 4)) {
+    if (!x3f_get_fake_camf_matrix(x3f, name, image, rect))
+      return 0;
+  } else {
+    if (!x3f_get_camf_matrix(x3f, name, 4, 0, 0, M_UINT, rect))
+      return 0;
+  }
+
```

Comment:
What you could do is refactor x3f_get_camf_rect, moving the transformation code (line 155 and onwards) to a separate function that takes a rectangle as a parameter. That function could then by used both by x3f_get_camf_rect and a new function that obtains rectangles from column ranges.